extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java

```diff
@@ -0,0 +1,251 @@
+package io.quarkus.devui.deployment.menu;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.quarkus.deployment.IsDevelopment;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.BuildSteps;
+import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
+import io.quarkus.devui.deployment.InternalPageBuildItem;
+import io.quarkus.devui.spi.buildtime.BuildTimeActionBuildItem;
+import io.quarkus.devui.spi.page.Page;
+import io.quarkus.devui.spi.workspace.Action;
+import io.quarkus.devui.spi.workspace.ActionBuilder;
+import io.quarkus.devui.spi.workspace.Display;
+import io.quarkus.devui.spi.workspace.DisplayType;
+import io.quarkus.devui.spi.workspace.Patterns;
+import io.quarkus.devui.spi.workspace.WorkspaceActionBuildItem;
+import io.quarkus.devui.spi.workspace.WorkspaceBuildItem;
+
+/**
+ * This creates the workspace Page
+ */
+@BuildSteps(onlyIf = IsDevelopment.class)
+public class WorkspaceProcessor {
+
+    @BuildStep
+    void locateWorkspaceItems(BuildSystemTargetBuildItem buildSystemTarget,
+            BuildProducer<WorkspaceBuildItem> workspaceProducer) {
+
+        Path outputDir = buildSystemTarget.getOutputDirectory();
+        Path projectRoot = outputDir.getParent();
+
+        List<WorkspaceBuildItem.WorkspaceItem> workspaceItems = new ArrayList<>();
```

Comment:
I'm worried about the memory cost when many files (like npm packages) exist.

Also, why do we need the list of files? What happens if a file is created or deleted?

Reply:
Ok I have now added config option to ignore folders or files, and by default also ignore node_modules. If a files is added or deleted, a hot reload will pick it up.